### PR TITLE
Miroil add event builder

### PR DESCRIPTION
**Pull request by erlend-g**
In Qtmir you have the class: qtmir:EventBuilder which contains both links to Qt and internal Mir stuff. See:
https://github.com/ubports/qtmir/blob/xenial_-_edge_-_wayland_-_x11/src/platforms/mirserver/eventbuilder.cpp

So what I have done is to split this class in two:
1. A base class in miroil that create MirEvents and deals with cookies.
2. A derived class that handles all the Qt stuff in (QtMir).

### CODE REVIEW

**Review by wmww — COMMENTED** [review]
This looks reasonable to me

**Comment by AlanGriffiths** [comment]
@erlend-g not directly relevant to this PR but, apparently, the reference version of QtMir should be https://gitlab.com/ubports/core/qtmir.

See: https://github.com/MirServer/mir/issues/1744#issuecomment-700336754

(I've only just noticed this while reviewing Mir issues)

File: include/miroil/miroil/event_builder.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        float pressure_value, float touch_major_value, float touch_minor_value, float size_value);
+    
+    // Key event
+    mir::EventUPtr make_event(MirInputDeviceId device_id, std::chrono::nanoseconds timestamp,
+        std::vector<uint8_t> const& cookie, MirKeyboardAction action, xkb_keysym_t key_code,
+        int scan_code, MirInputEventModifiers modifiers);
+
+    // Touch event
+    mir::EventUPtr make_event(MirInputDeviceId device_id, std::chrono::nanoseconds timestamp,
+        std::vector<uint8_t> const& mac, MirInputEventModifiers modifiers);
```
Comment (AlanGriffiths) [inline include/miroil/miroil/event_builder.h:75]: I know the mirclient API has this horrible abuse of overloading `make_event()`, but I don't think it right to repeat the mistake in miroil. Would it be painful in the QtMir code to have `make_touch_event()`, `make_key_event()`, etc?

Comment (erlend-g) [reply include/miroil/miroil/event_builder.h:75]: Yes, that sound like a good idea and is quick to do. So I'll change the names.

Comment (erlend-g) [reply include/miroil/miroil/event_builder.h]: Fixed that, and talk to Marius Gripsgard about branches... So https://github.com/ubports/qtmir/tree/xenial_-_edge_-_wayland_-_x11it the correct one to use for now.

**Review by AlanGriffiths — APPROVED** [review]

**Pull request merged**